modules/fedimint-lnv2-client/src/api.rs

```diff
@@ -24,6 +26,8 @@ pub trait LnFederationApi {
         &self,
         contract_id: &ContractId,
     ) -> FederationResult<Option<u64>>;
+
+    async fn fetch_gateways(&self) -> FederationResult<Vec<SafeUrl>>;
```

Comment:
I don't see client calls to add and remove. 

Consider following up with:

* these client calls
* command line handling so that `fedimint-cli module lnv2 vetted add <url>`, `fedimint-cli module lnv2 vetted list` can be already used.
* sanity test

All of these are done in meta module, so you might copy&paste&modify if it saves you any time.

Reply:
I just pushed the first two changes a a commit, however I am unsure how to factor it. Initially I extended the LnFederationApi with the add and remove calls, however so far they are currently only used in the cli which is only compiled conditionally, similar situation with the admin_auth method on the client module itself like in the meta module but here clippy actually errored. I am wondering, how will the UI call these methods and If there is a better way to factor this. I have to little context here to decide this.

Tests can be done In a follow up.